### PR TITLE
added: max character length plugin for comments & publication

### PR DESCRIPTION
**Pull request by harryranakl**
## What does this PR do?
#401 - Limit to 10k characters for posts and comments to utilise free node2 bundlr

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

## How should this be tested?
ezgif com-gif-maker (1)

## Checklist
- My code doesn't follow the style guidelines of this project

### CODE REVIEW

**Comment by bigint** [comment]
cc @harish-sethuraman for lexical 🙏🏼

File: packages/data/constants.ts [hunk 1]
```diff
@@ -121,3 +121,7 @@ export const LS_KEYS = {
 export const S3_BUCKET = {
   LENSTER_MEDIA: 'lenster-media'
 };
+
+// lexical MaxLengthPlugin
+export const CHRACTERS_MAXLENGTH = 10000;
```
Comment (harish-sethuraman) [inline packages/data/constants.ts:126]:
```suggestion
export const CHARACTERS_MAXLENGTH = 10000;
```

File: packages/data/constants.ts [hunk 2]
```diff
@@ -121,3 +121,7 @@ export const LS_KEYS = {
...
+export const SHOW_REMAINING_CHRACTERS_LENGTH = 20;
```
Comment (harish-sethuraman) [inline packages/data/constants.ts:127]:
```suggestion
export const SHOW_REMAINING_CHARACTERS_LENGTH = 20;
```

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 3]
```diff
@@ -527,22 +530,28 @@ const NewPublication: FC<Props> = ({ publication }) => {
...
-              )
-            }
-            onClick={createPublication}
-          >
-            {isComment ? 'Comment' : 'Post'}
-          </Button>
+        <div className="ml-auto pt-2 sm:pt-2 space-x-4">
+          <div className="flex items-center space-x-4">
+            <MaxLengthPlugin
+              maxLength={CHRACTERS_MAXLENGTH}
```
Comment (harish-sethuraman) [inline apps/web/src/components/Composer/NewPublication.tsx:536]:
```suggestion
              maxLength={CHARACTERS_MAXLENGTH}
```

File: apps/web/src/components/Composer/NewPublication.tsx [hunk 4]
```diff
@@ -527,22 +530,28 @@ const NewPublication: FC<Props> = ({ publication }) => {
...
+              remainingCharactersLength={SHOW_REMAINING_CHRACTERS_LENGTH}
```
Comment (harish-sethuraman) [inline apps/web/src/components/Composer/NewPublication.tsx:537]:
```suggestion
              remainingCharactersLength={SHOW_REMAINING_CHARACTERS_LENGTH}
```

**Comment by harish-sethuraman** [comment]
Screenshot 2022-12-29 at 12 32 40 PM
Something wrong with me? Im not able to see the limit itself? I used this https://web-i6zkogthf-lenster.vercel.app/ preview deployment

**Comment by harryranakl** [comment]
in the case of typing content the round box shows the remaining characters but didn't show for copy and paste content
updated for the copy-and-paste scenario.

Screenshot 2022-12-30 at 12 48 36 AM

**Comment by bigint** [comment]
I think it would be great to show the remaining counter count only after reaching near the limit. I also go with circular progress bar instead of counter

**Pull request closed without merging**